x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/authorization/ElasticInferenceServiceAuthorizationHandler.java

```diff
@@ -0,0 +1,136 @@
+    }
+
+    // only use for testing
+    ElasticInferenceServiceAuthorizationHandler(@Nullable String baseUrl, ThreadPool threadPool, Logger logger) {
+        this.baseUrl = baseUrl;
+        this.threadPool = Objects.requireNonNull(threadPool);
+        this.logger = Objects.requireNonNull(logger);
+    }
+
+    /**
+     * Retrieve the authorization information from EIS
+     * @param listener a listener to receive the response
+     * @param sender a {@link Sender} for making the request to the EIS gateway
+     */
+    public void getAuthorization(ActionListener<ElasticInferenceServiceAuthorization> listener, Sender sender) {
+        try {
+            logger.debug("Retrieving authorization information from the Elastic Inference Service.");
+
+            if (Strings.isNullOrEmpty(baseUrl)) {
+                logger.warn("The base URL for the authorization service is not valid, rejecting authorization.");
+                listener.onResponse(ElasticInferenceServiceAuthorization.newDisabledService());
+                return;
+            }
+
+            // ensure that the sender is initialized
+            sender.start();
+
+            ActionListener<InferenceServiceResults> newListener = ActionListener.wrap(results -> {
+                if (results instanceof ElasticInferenceServiceAuthorizationResponseEntity authResponseEntity) {
+                    listener.onResponse(ElasticInferenceServiceAuthorization.of(authResponseEntity));
+                } else {
+                    logger.warn(
+                        Strings.format(
+                            FAILED_TO_RETRIEVE_MESSAGE + " Received an invalid response type: %s",
+                            results.getClass().getSimpleName()
+                        )
+                    );
+                    listener.onResponse(ElasticInferenceServiceAuthorization.newDisabledService());
+                }
+                requestCompleteLatch.countDown();
+            }, e -> {
+                Throwable exception = e;
+                if (e instanceof ElasticsearchWrapperException wrapperException) {
+                    exception = wrapperException.getCause();
+                }
+
+                logger.warn(Strings.format(FAILED_TO_RETRIEVE_MESSAGE + " Encountered an exception: %s", exception));
+                listener.onResponse(ElasticInferenceServiceAuthorization.newDisabledService());
+                requestCompleteLatch.countDown();
+            });
+
+            var request = new ElasticInferenceServiceAuthorizationRequest(baseUrl, getCurrentTraceInfo());
+
+            sender.sendWithoutQueuing(logger, request, AUTH_RESPONSE_HANDLER, DEFAULT_TIMEOUT, newListener);
+        } catch (Exception e) {
+            logger.warn(Strings.format("Retrieving the authorization information encountered an exception: %s", e));
+        }
+    }
+
+    // TODO do we need this?
+    private TraceContext getCurrentTraceInfo() {
```

Comment:
@maxjakob @timgrein Do we need trace info for authorization calls?

Reply:
Are the latencies and response code monitored otherwise (like APM)? If so, I think we don't need tracing for this one. The call ends in EIS with a hashmap lookup.

Reply:
> Are the latencies and response code monitored otherwise (like APM)?

hmm no I don't think we monitor them. Maybe we should leave it in then?

Reply:
I would say let's leave it in and propagate through the authorization request, it doesn't hurt to have it, right? @maxjakob do we explicitly attach to this trace on the EIS side?

Reply:
If this is the only observability we have on these endpoint calls it would indeed be good to leave it in.

We do not explicitly attach to this trace in EIS yet but could do. Also planning to add APM monitoring on that side.

Reply:
Yeah sounds good I'll leave it in and remove the todo 👍